File: codewars/7kyu/balanced-number-special-numbers-series-#1/dduck-balanced-number-special-numbers-series-#1.py
# https://www.codewars.com/kata/5a4e3782880385ba68000018

from unittest import TestCase


def balanced_num(number):
    digits = [int(x) for x in str(number)]
    half_len = int((len(digits) - 2) / 2 if len(digits) % 2 == 0 else (len(digits) - 1) / 2)
    left_sum = sum(digits[:half_len])
    right_sum = sum(digits[-half_len:])

    if half_len == 0 or left_sum == right_sum:
        return "Balanced"
    return "Not Balanced"


TestCase().assertEqual(balanced_num(7), "Balanced");
TestCase().assertEqual(balanced_num(959), "Balanced");
TestCase().assertEqual(balanced_num(13), "Balanced");
TestCase().assertEqual(balanced_num(432), "Not Balanced");
TestCase().assertEqual(balanced_num(424), "Balanced");
TestCase().assertEqual(balanced_num(1024), "Not Balanced")
TestCase().assertEqual(balanced_num(66545), "Not Balanced")
TestCase().assertEqual(balanced_num(295591), "Not Balanced")
TestCase().assertEqual(balanced_num(1230987), "Not Balanced")
TestCase().assertEqual(balanced_num(56239814), "Balanced")

File: codewars/7kyu/balanced-number-special-numbers-series-#1/dduck-balanced-number-special-numbers-series-#1.js
// https://www.codewars.com/kata/5a4e3782880385ba68000018

const sum = (arr) => arr.reduce((a, b) => a + b, 0);
const balancedNum = (number) => {
  const splitedNum = String(number).split('').map(Number);
  const halfLength = Math.floor(splitedNum.length / 2);
  if (sum(splitedNum.slice(0, halfLength)) === sum(splitedNum.slice(halfLength + 1))) {
    return 'Balanced';
  }
  return 'Not Balanced';
};

console.log(balancedNum(7), 'Balanced');
console.log(balancedNum(959), 'Balanced');
console.log(balancedNum(13), 'Balanced');
console.log(balancedNum(432), 'Not Balanced');
console.log(balancedNum(424), 'Balanced');
console.log(balancedNum(1024), 'Not Balanced');
console.log(balancedNum(66545), 'Not Balanced');
console.log(balancedNum(295591), 'Not Balanced');
console.log(balancedNum(1230987), 'Not Balanced');
console.log(balancedNum(56239814), 'Balanced');
